Exercises/4-count-types.js

```diff
@@ -1,5 +1,14 @@
 'use strict';
 
-const countTypesInArray = null;
-
+const countTypesInArray  = arg => {
+  const result = {};
+  for (const value of arg) {
+    const field = typeof value;
+    if (field in result)
+      result[field]++;
+    else result[field] = 1;
```

Comment:
```js
if (field in result) result[field]++;
else result[field] = 1;
```
or
```js
const counter = result[field];
if (counter) result[field] = ++counter;
else result[field] = 1;
```
or
```js
let counter = result[field] || 0;
counter++;
result[field] = counter;
```
or
```js
const counter = result[field] || 0;
result[field] = counter + 1;
```

Reply:
Second option doesn't work as counter is const

Reply:
I don't understand second code:

const counter = result[field];
if (counter) result[field] = ++counter;
else result[field] = 1;_

What it's mean  in "if" and after "if" ?

Reply:
If there is no field (field) in Your result, then counter is undefined, so it will be evaluated as false program will go to else statement and will add field (field) and assign value '1' to it. In case there is field (field) any value other than '0', ' "" ' (empty string) will be evaluated as true and than programm will assign to field (field) of result object value of incremented counter

Reply:
Thanks)

Reply:
Thanks for your advices!